File: 20250315/leetcode3110. 字符串的分数/main.js
// 给你一个字符串 s 。一个字符串的 分数 定义为相邻字符 ASCII 码差值绝对值的和。

// 请你返回 s 的 分数 。

/**
 * @param {string} s
 * @return {number}
 */
var scoreOfString = function (s) {
  const n = s.length;
  let res = 0;
  for (let i = 1; i < n; i++) {
    res += Math.abs(s[i].charCodeAt() - s[i - 1].charCodeAt());
  }
  return res;
};

File: 20250315/leetcode3110. 字符串的分数/main.cpp

#include <iostream>
#include <cmath>
using namespace std;

class Solution
{
public:
  int scoreOfString(string s)
  {
    const int n = s.size();
    int res = 0;
    for (int i = 1; i < n; i++)
    {
      res += abs(s[i] - s[i - 1]);
    }
    return res;
  }
};

int main()
{
  Solution solution;
  cout << solution.scoreOfString("leetcode");
  return 0;
}